File: java/string/657_RobotReturnOrigin.java
class Solution {
    public boolean judgeCircle(String moves) {
        return count(moves,"U") == count(moves, "D") &&
            count(moves,"L") == count(moves,"R");
    }

    public int count(String moves, String s){
        return moves.length() - moves.replaceAll(s,"").length();
    }
}

File: Leetcode/C&CPP/string/657_RobotReturnOrigin.c
bool judgeCircle(char* moves) {
    int x = 0, y = 0;
    for (; *moves; moves++){
        switch(*moves){
        case 'U': y+=1; break;
        case 'D': y-=1; break;
        case 'L': x-=1; break;
        case 'R': x+=1; break;
        default: return 0;
        }
    }
    return x==0 && y==0;
}
